hwconf/Zerno/hw_zerno_drive_core.c

```diff
@@ -36,727 +36,810 @@
-	mcconf->foc_f_zv = 10000.0;
-	mcconf->foc_current_kp = 0.01;
-	mcconf->foc_current_ki = 10.0;
-	mc_interface_set_configuration(mcconf);
+    is_encoder_done = true;
+}
 
-	float current = 2.0;
-	float offset = 0.0;
-	float ratio = 0.0;
-	bool inverted = false;
+bool is_hw_fault( void )
+{
+    bool custom_fault = false;
 
-	mcpwm_foc_encoder_detect(current, false, &offset, &ratio, &inverted);
+    //TODO: Add a custom fault here.
 
-	mcconf_old->foc_encoder_offset = offset;
-	mcconf->foc_encoder_offset = offset;
+    return( custom_fault );
+}
 
-	mc_interface_set_configuration(mcconf);
-	mc_interface_set_configuration(mcconf_old);
+static void adc_read_callback( void )
+{
+    float filter_knob = 0.0;
 
-	mempools_free_mcconf(mcconf);
-	mempools_free_mcconf(mcconf_old);
+    filter_knob = ADC_VOLTS( ADC_IND_EXT );
 
-	is_encoder_done = true;
+    UTILS_LP_FAST( knob_read_in_volts, filter_knob, 0.01 );
 }
 
-bool is_hw_fault(void) {
-	bool custom_fault = false;
+float get_pfc_temp( void )
+{
+    static float temp_pfc_filtered = 0.0;
 
-	//TODO: Add a custom fault here.
+    float temp_pfc = ( 1.0 / ( ( logf( NTC_RES( ADC_Value[ ADC_IND_TEMP_PFC ] ) / 10000.0 ) / 3455.0 ) + ( 1.0 / 298.15 ) ) - 273.15 );
 
-	return (custom_fault);
+    UTILS_LP_FAST( temp_pfc_filtered, temp_pfc, 0.1 );
+    return temp_pfc_filtered;
 }
 
-static void adc_read_callback(void) {
+static void terminal_print_info( int argc,
+                                 const char ** argv )
+{
+    ( void ) argc;
+    ( void ) argv;
+
+    eeprom_var data_stored;
+    eeprom_var check_cal;
+
+    conf_general_read_eeprom_var_hw( &data_stored, EEPROM_ADDR_ENCODER_VALUE );
+    commands_printf( "Encoder stored value: %f", ( double ) ( data_stored.as_float ) );
+    conf_general_read_eeprom_var_hw( &check_cal, EEPROM_ADDR_CALIBRATION_CHECK );
+    commands_printf( "Calibration status: %d", check_cal.as_i32 );
+
+    ( is_pfc_ok() ) ? commands_printf( "PFC:OK" ) : commands_printf( "PFC:OFF" );
+
+    commands_printf( "ADC: %f", ( double ) knob_read_in_volts );
+    commands_printf( "ADC_cal: %f", ( double ) encoder_calibrated_value_in_volts );
+    commands_printf( "Encoder min: %f", ( double ) encoder_min_value_in_volts );
+    commands_printf( "min cal: %f", ( double ) encoder_min_calibrated_value );
+    commands_printf( "linear: %f", ( double ) knob_index );
+    commands_printf( "step: %f", ( double ) steps );
+    commands_printf( "speed (eRPM): %f", ( double ) speed_erpm_setpoint );
+    commands_printf( "speed (RPM): %f", ( double ) ( speed_erpm_setpoint / 4 ) );
+}
 
-	float filter_knob = 0.0;
+float get_knob_read( void )
+{
+    return( encoder_calibrated_value_in_volts );
+}
 
-	filter_knob = ADC_VOLTS(ADC_IND_EXT);
+static void terminal_motor_run( int argc,
+                                const char ** argv )
+{
+    ( void ) argc;
+    ( void ) argv;
+
+    if( strcmp( argv[ 1 ], "ON" ) == 0 )
+    {
+        commands_printf( "Motor ON" );
+        motor_start = true;
```

Comment:
`motor_start` is only used on this tiny implementation. Better to declare it inside this function with `static`.